pdns/logger.cc

```diff
@@ -67,42 +68,47 @@ void Logger::log(const string& msg, Urgency u) noexcept
       }
     }
 
-    string prefix;
+    string severity;
     if (d_prefixed) {
       switch (u) {
       case All:
-        prefix = "[all] ";
+        severity = "All";
         break;
       case Alert:
-        prefix = "[ALERT] ";
+        severity = "Alert";
         break;
       case Critical:
-        prefix = "[CRITICAL] ";
+        severity = "Critical";
         break;
       case Error:
-        prefix = "[ERROR] ";
+        severity = "Error";
         break;
       case Warning:
-        prefix = "[WARNING] ";
+        severity = "Warning";
         break;
       case Notice:
-        prefix = "[NOTICE] ";
+        severity = "Notice";
         break;
       case Info:
-        prefix = "[INFO] ";
+        severity = "Info";
         break;
       case Debug:
-        prefix = "[DEBUG] ";
+        severity = "Debug";
         break;
       case None:
-        prefix = "[none] ";
+        severity = "None";
         break;
       }
     }
 
     static std::mutex m;
     std::lock_guard<std::mutex> l(m); // the C++-2011 spec says we need this, and OSX actually does
-    clog << string(buffer) + prefix + msg << endl;
+    if (d_prefixed) {
+      clog << string(static_cast<char*>(&buffer[0])) << "msg=" << std::quoted(msg) << " prio=" << std::quoted(severity) << endl;
```

Comment:
Probably more nice to change buffer to:
```
    std::array<char, 50> buffer{};
    buffer[0] = '\0';
```
and then use `buffer.size()` and `buffer.data()` in appropriate places

Reply:
Also, `msg` is glued to prefix.

Reply:
I am also wondering if we should keep using a single string, so that the actual write happens in one syscall.

Reply:
I'm doing a few tests to see if the number of syscalls per log line is changing

Reply:
```
99113 pdns_recursor CALL  write(2,0x1f972cf9780,0x90)
 99113 pdns_recursor GIO   fd 2 wrote 144 bytes
       "Aug 01 09:21:59 msg="Listening for HTTP requests" subsystem="webserver\
        " level="0" prio="Info" tid="0" ts="1690874519.656" address="0.0.0.0:8\
        083""
 99113 pdns_recursor RET   write 144/0x90
 99113 pdns_recursor CALL  write(2,0x1f8adb46b8f,0x1)
 99113 pdns_recursor GIO   fd 2 wrote 1 bytes
       "
       "
 99113 pdns_recursor RET   write 1
```

With the PR:
```2629 pdns_recursor CALL  write(2,0xa365443aa51,0x10)
  2629 pdns_recursor GIO   fd 2 wrote 16 bytes
       "Aug 01 09:24:04 "
  2629 pdns_recursor RET   write 16/0x10
  2629 pdns_recursor CALL  write(2,0xa3623d3f0c0,0x80)
  2629 pdns_recursor CALL  kbind(0x79387a645d78,24,0xc7ca0038866b7d4a)
  2629 pdns_recursor CALL  kbind(0xa36c599a6e8,24,0xc7ca0038866b7d4a)
  2629 pdns_recursor GIO   fd 2 wrote 128 bytes
       "msg="Listening for HTTP requests" subsystem="webserver" level="0" prio\
        ="Info" tid="0" ts="1690874644.702" address="0.0.0.0:8083""
  2629 pdns_recursor RET   kbind 0
  2629 pdns_recursor RET   write 128/0x80
  2629 pdns_recursor CALL  kbind(0x79387a645d78,24,0xc7ca0038866b7d4a)
  2629 pdns_recursor RET   kbind 0
  2629 pdns_recursor CALL  write(2,0xa365443a93f,0x1)
  2629 pdns_recursor GIO   fd 2 wrote 1 bytes
       "
       "
  2629 pdns_recursor RET   write 1
 ```

So the line itself is split with this PR into a separate write for the timestamp and the msg. The end-of-line handling did not change (but we might want to change that so it's all in a single write)

Reply:
thanks for pushing this change!